x.py

```diff
@@ -116,14 +119,43 @@ def build(dir: str, jobs: int, ghproxy: bool, ninja: bool, unittest: bool, compi
         target.append("unittest")
     run(cmake, "--build", ".", f"-j{jobs}", "-t", *target, verbose=True, cwd=dir)
 
-def cpplint() -> None:
-    command = find_command("cpplint", msg="cpplint is required")
-    options = ["--linelength=120", "--filter=-build/include_subdir,-legal/copyright,-build/c++11"]
-    sources = [*glob("src/*.h"), *glob("src/*.cc")]
-    run(command, *options, *sources, verbose=True)
+def get_source_files() -> List[str]:
+    return [
+        *glob("src/**/*.h", recursive=True),
+        *glob("src/**/*.cc", recursive=True),
+        *glob("tests/cppunit/**/*.h", recursive=True),
+        *glob("tests/cppunit/**/*.cc", recursive=True),
+        *glob("tools/kvrocks2redis/**/*.h", recursive=True),
+        *glob("tools/kvrocks2redis/**/*.cc", recursive=True),
+    ]
+
+def check_format(clang_format_path: str, fix: bool = False) -> None:
+    command = find_command(clang_format_path, msg="clang-format is required")
+
+    version_res = run_pipe(command, '--version').read().strip()
+    version_str = re.search(r'version\s+((?:\w|\.)+)', version_res).group(1)
+
+    version = check_version(version_str, CLANG_FORMAT_REQUIRED_VERSION, "clang-format")
+
+    if version[0] > 12:
+        raise RuntimeError("We use clang-format 12 in CI,\n"
+            "  so we recommend that you also use this version locally to avoid inconsistencies.\n"
+            "  You can install it from your package manager (usually in clang-12 package)\n"
+            "  or download it from https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1")
```

Comment:
```suggestion
        raise RuntimeError("We use clang-format 12 in CI,"
            "  so we recommend that you also use this version locally to avoid inconsistencies."
            "  You can install it from your package manager (usually in clang-12 package)"
            "  or download it from https://github.com/llvm/llvm-project/releases/tag/llvmorg-12.0.1")
```

I suggest not to manually wrap lines.

Reply:
done